src/components/input/inputFileAvatar/inputFileAvatar.tsx

```diff
@@ -0,0 +1,184 @@
+        event.stopPropagation();
+        setImagePreview(null);
+        setSelectState(SelectState.IDLE);
+        if (imagePreview) {
+            URL.revokeObjectURL(imagePreview);
+        }
+    };
+
+    const selecterVariantStateClasses = classNames(
+        'group flex size-16 items-center justify-center rounded-full bg-neutral-0 hover:shadow-neutral focus:outline-none focus-visible:ring  focus-visible:ring-offset', // Default
+        {
+            ['border-[1px] border-neutral-100 hover:border-neutral-200 border-dashed cursor-pointer focus-visible:ring-primary']:
+                selectState === SelectState.IDLE, // IDLE State
+            ['cursor-default border-[1px] border-primary-400 focus-visible:ring-primary']:
+                selectState === SelectState.SELECTING, // SELECTING State
+            ['cursor-default border-[5px] border-neutral-100 border-dashed focus-visible:ring-primary']:
+                selectState === SelectState.SUCCESS, // SUCCESS State
+            ['border-[1px] border-critical-500 hover:border-critical-600 border-dashed cursor-pointer focus-visible:ring-critical']:
+                selectState === SelectState.ERROR, // ERROR State
+        },
+    );
+
+    return (
+        <>
+            <InputContainer id={id} alert={selectError} invisible {...otherContainerProps}>
+                <div {...getRootProps()} className={selecterVariantStateClasses} aria-label="Select File">
+                    <input {...getInputProps()} id={id} type="file" aria-label="Avatar Image Select" />
+                    {selectState === SelectState.SUCCESS && imagePreview ? (
+                        <div className="relative">
+                            <Avatar src={imagePreview} size="lg" className="cursor-pointer" data-testid="avatar" />
+                            <button
+                                onClick={handleCancel}
+                                className={classNames(
+                                    'absolute -right-1 -top-1 cursor-pointer rounded-full bg-neutral-0 p-1 shadow-neutral focus:outline-none focus-visible:ring focus-visible:ring-primary focus-visible:ring-offset',
+                                )}
+                                aria-label="Cancel Selection"
+                            >
+                                <Icon icon={IconType.CLOSE} size="sm" />
+                            </button>
+                        </div>
+                    ) : (
+                        <>
+                            {selectState === SelectState.IDLE && (
+                                <Icon
+                                    icon={IconType.ADD}
+                                    size="lg"
+                                    className="text-neutral-400 group-hover:text-neutral-600"
+                                />
+                            )}
+                            {selectState === SelectState.SELECTING && <Spinner size="lg" variant="neutral" />}
```

Comment:
We're never setting the state to `SELECTING` currently, right? We should probably set it before we get the image preview